### PR TITLE
WIP: Add constant errors

### PR DESCRIPTION
**Pull request by alejandrojnm**
Added the constant errors to the lib, It is common for a package to define some of the errors that it may raise so that the user can match against

### CODE REVIEW

**Comment by alejandrojnm** [comment]
with this code, you can see this error:
```golang
configK8s := &civogo.KubernetesClusterConfig{
	NumTargetNodes: 0,
	Name: "rrr",
}
//
resp, err := client.NewKubernetesClusters(configK8s)
if err != nil {
	fmt.Println(err)
}
```
`DatabaseKubernetesClusterInvalid: The Kubernetes cluster could not be created due to it failing validation, Num target nodes is not included in the list`

if you user a existent name you will get this error:
```golang
configK8s := &civogo.KubernetesClusterConfig{
	NumTargetNodes: 5,
	Name: "existent-name",
}
//
resp, err := client.NewKubernetesClusters(configK8s)
if err != nil {
	fmt.Println(err)
}
```
`DatabaseKubernetesClusterDuplicate: Failed to create a Kubernetes cluster, one already exists with that name`

you can use **errors.Is()** in this way:

```golang
if err != nil {
     if errors.Is(err, civogo.DatabaseKubernetesClusterDuplicateError) {
	 // take some actions
     }
}
```

**Comment by andyjeffries** [comment]
Happy with this approach. Please continue to create all the errors. Also ensure this is documented some way for users of the library. Thanks.

**Comment by alejandrojnm** [comment]
Ok, i will finish the rest of the error

File: errors.go [hunk 1]
```diff
@@ -80,9 +291,550 @@ func decodeERROR(err error) error {
 	}
 
 	switch dat["code"] {
+	// Kuberenetes Error
```
Comment (andyjeffries) [inline errors.go:294]: What is `Kuberenetes`? ;-)

**Comment by andyjeffries** [comment]
Happy with this. Just the typo in a comment, then we can release it. Is there anything in the README about error handling? Either way, I think we should add some documentation to explain how this method works (like you did above).

**Pull request merged**